notification/views.py

```diff
@@ -29,22 +35,23 @@ def filter_queryset(self, queryset):
 
         return qs
 
-    # Change notification to seen once requested
-    def finalize_response(self, request, *args, **kwargs):
-        result = super().finalize_response(request, *args, **kwargs)
-
-        if not getattr(request, 'child_route', False):
-            queryset = self.paginate_queryset(
-                self.filter_queryset(self.get_queryset()))
-            Notification.objects\
-                .filter(id__in=[q.id for q in queryset])\
-                .update(status=Notification.STATUS_SEEN)
-
-        return result
+    @list_route(permission_classes=[permissions.IsAuthenticated],
+                methods=['put'],
+                url_path='status')
+    def status_update(self, request, version=None):
+        serializer = NotificationStatusSerializer(data=request.data, many=True)
+        valid = serializer.is_valid()
+        if not valid:
+            raise exceptions.ValidationError(serializer.errors)
+        for item in serializer.data:
```

Comment:
If this was NotificationSerailizer, then I think the a simple `serializer.save()` would do instead of this for loop.